### PR TITLE
[To review] Rnn - seq2seq add math formula

### PR DESCRIPTION
**Pull request by goldmermaid**

### CODE REVIEW

**Comment by goldmermaid** [comment]
Check passed.

File: chapter_recurrent-neural-networks/seq2seq.md [hunk 1]
```diff
@@ -50,14 +65,27 @@ encoder = Seq2SeqEncoder(vocab_size=10, embed_size=8,
...
+```{.python .input}
+len(state), state[0].shape, state[1].shape
 ```
 
 ## Decoder
 
-We directly use the hidden state of the encoder in the final time step as the initial hidden state of the decoder. This requires that the encoder and decoder RNNs have the same numbers of layers and hidden units.
+As we just introduced, the content vector $\mathbf{c}$ encodes the information from the whole input sequence $x_1, \ldots, x_T$. Suppose that the given outputs in the training set are $y_1, \ldots, y_{T'}$. At each timestep $t'$, the conditional probability of output $y_{t'}$ will base on the previous output sequences $y_1, \ldots, y_{t'-1}$ and the content vector $\mathbf{c}$, i.e., $$P(y_{t'} \mid y_1, \ldots, y_{t'-1}, \mathbf{c}).$$
+
+Hence, we can use another RNN as the decoder. At timestep $t'$, the decoder will update its hidden state $\mathbf{s_{t'}}$ by three inputs: $y_{t'-1}$'s feature vector $\mathbf{y_{t'-1}}$, the content vector $\mathbf{c}$, and the hidden state of the last timestep $\mathbf{s_{t'}}$. Let us denote the transformation of the RNN's hidden states within the decoder by a function $g$:
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/seq2seq.md]: last timestep $\mathbf{s_{t'}}$ is wrong, should be t'-1

File: chapter_recurrent-neural-networks/seq2seq.md [hunk 2]
```diff
@@ -50,14 +65,27 @@ encoder = Seq2SeqEncoder(vocab_size=10, embed_size=8,
...
 ```
 
 ## Decoder
 
-We directly use the hidden state of the encoder in the final time step as the initial hidden state of the decoder. This requires that the encoder and decoder RNNs have the same numbers of layers and hidden units.
+As we just introduced, the content vector $\mathbf{c}$ encodes the information from the whole input sequence $x_1, \ldots, x_T$. Suppose that the given outputs in the training set are $y_1, \ldots, y_{T'}$. At each timestep $t'$, the conditional probability of output $y_{t'}$ will base on the previous output sequences $y_1, \ldots, y_{t'-1}$ and the content vector $\mathbf{c}$, i.e., $$P(y_{t'} \mid y_1, \ldots, y_{t'-1}, \mathbf{c}).$$
+
+Hence, we can use another RNN as the decoder. At timestep $t'$, the decoder will update its hidden state $\mathbf{s_{t'}}$ by three inputs: $y_{t'-1}$'s feature vector $\mathbf{y_{t'-1}}$, the content vector $\mathbf{c}$, and the hidden state of the last timestep $\mathbf{s_{t'}}$. Let us denote the transformation of the RNN's hidden states within the decoder by a function $g$: 
+
+$$\mathbf{s_{t'}} = f (\mathbf{y_{t'-1}}, \mathbf{c}, \mathbf{s_{t'-1}}).$$
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/seq2seq.md]: f -> g

File: chapter_recurrent-neural-networks/seq2seq.md [hunk 3]
```diff
@@ -20,7 +20,22 @@ npx.set_np()
 
 ## Encoder
 
-In the encoder, we use the word embedding layer to obtain a feature index from the word index of the input language and then input it into a multi-level LSTM recurrent unit. The input for the encoder is a batch of sequences, which is 2-D tensor with shape (batch size, sequence length). It outputs both the LSTM outputs, e.g the hidden state, for each time step and the hidden state and memory cell of the last time step.
+Recall that the encoder of seq2seq can transform the inputs with different length to a universal fixed length content vector $\mathbf{c}$ by encoding the sequence information inside $\mathbf{c}$. We usually use RNN layers within the encoder.
+
+Let us think of an input sequence $x_1, \ldots, x_T$, where $x_t$ is $t$-th word of the inputs. At timestep $t$, the RNN will have two vectors as input: the $x_t$'s feature vector $\mathbf{x_t}$ and the hidden state of the last timestep $\mathbf{h_t}$. Let us denote the transformation of the RNN's hidden states by a function $f$:
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/seq2seq.md]: the hidden state of the last timestep $\mathbf{h_t}$
is wrong,
should be t-1

**Comment by astonzhang** [comment]
Please see my fix:
https://github.com/d2l-ai/d2l-en/commit/f5488f1e02ee9b5a9231621d5ee5bc55d07b8a9b

**Pull request merged**